use client-go v2.0 and upgrade k8s v1.5

ngtuna opened the pull request:
This PR I try upgrading k8s from 1.4 to 1.5.5 and using client-go v2.0 instead of the build-in client pkg of k8s. The [compatibility matrix](https://github.com/kubernetes/client-go/tree/master#compatibility-matrix) figured out they match exactly. By doing this, all k8s objects are no longer being imported from k8s pkgs but we will start using client-go as the main library.

Few more things need to be improved:

- exec, port-forward commands still being wrapped by kubectl pkg. Client-go doesn't support it well.
- double check the glide vendor tree to remove more unused libs.

anguslees commented:
I see you've combined this with renaming create->deploy, etc.  Fine by me, but it should be mentioned in the commit description (and makes it harder to review).  The diff is now big enough to crash my chromium tab :)

ngtuna commented:
@anguslees Thanks for mentioning that. Totally agree. Here below I am listing out the changes I made in the PR:

- included client-go v2.0 and k8s v1.5.5 with glide. Check glide.yaml for detailed packages.
- renamed create.go to deploy.go
- added comments to all necessary functions
- replaced all api calls using client-go except kubectl port-forward (function call) and exec (topic related activities).
- kubectl port-forward (function call) and exec (topic related activities) also were updated to fit k8s v1.5
- by using client-go, we can get rid of kubectl proxy. So kubeless-controller pod now no longer contains kubectl container anymore.
- re-structured function spec to fit the new client-go custom object format.

The PR is ready for review now :-)

anguslees commented on pkg/utils/k8sutil.go:
```diff
@@ -17,115 +17,128 @@ limitations under the License.
+	}
+
+	return httpClient.Get(fmt.Sprintf("https://%s:%s/apis/k8s.io/v1/functions?watch=true&resourceVersion=%s",
+		host, port, resourceVersion))
```
This won't work when `host` is an IPv6 literal.  Use `net.JoinHostPort(host, port)` and/or construct a `url.URL` object and return `url.String()`  (the latter will also make sure that resourceVersion is quoted correctly, etc)

ngtuna commented:
@anguslees Yeah you're right. We should do that. I can quickly refactor this PR.

ngtuna commented on pkg/controller/controller.go:
```diff
@@ -252,16 +256,17 @@ func (c *Controller) monitor(watchVersion string) (<-chan *Event, <-chan error)
 				}
```
Fixed

ngtuna commented:
Thanks @anguslees. Let's go
